### PR TITLE
TC04 - "Login page displays when un-logged User clicks on Book ticket tab"

### PR DESCRIPTION
**Pull request by viethuynh140499**

### CODE REVIEW

File: src/main/java/common/helpers/BrowserHelper.java [hunk 1]
```diff
@@ -72,4 +68,16 @@ public static void returnPreviousPage() {
         JavascriptExecutor js = (JavascriptExecutor) driver;
         js.executeScript("window.history.go(-1)");
     }
+
+    public static String getTitle(){
+        return getDriver().getTitle();
+    }
+
+    public static void handleAlert(boolean action) {
```
Comment (thanhlelgg) [inline src/main/java/common/helpers/BrowserHelper.java:76]: action is not a good name for a boolean. action value is true or false doesn't make sense
you should you an enum for this and action could be either Accept or Dismiss

File: src/test/java/test_cases/LoginTests.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  public void TC01() {
+    User user = new User(Constant.USERNAME, Constant.PASSWORD);
+
+    homePage.clickLoginTab();
+    loginPage.login(user);
+
+    String actualResult = homePage.getWelcomeMessage();
+    String expectedResult = "Welcome " + user.getUsername();
+
+    Assert.assertEquals(actualResult, expectedResult);
```
Comment (thanhlelgg) [inline src/test/java/test_cases/LoginTests.java:26]: there should be failed message for all check point to describe what we're verifying

File: src/test/java/test_cases/LoginTests.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    String actualResult = loginPage.getErrorMessage();
+    String expectedResult = "There was a problem with your login and/or errors exist in your form.";
+
+    Assert.assertEquals(actualResult, expectedResult);
+  }
+
+  @Test(description = "User cannot log into Railway with invalid password ")
+  public void TC03() {
+
+    User user = new User(Constant.USERNAME, DataHelper.getRandomInvalidPassword());
```
Comment (thanhlelgg) [inline src/test/java/test_cases/LoginTests.java:45]: invalid password doesn't have to be random. Just use a constant value. It's easier to traceback if something went wrong

File: src/test/java/test_cases/LoginTests.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    String actualResult = loginPage.getErrorMessage();
+    String expectedResult = "There was a problem with your login and/or errors exist in your form.";
+
+    Assert.assertEquals(actualResult, expectedResult);
+  }
+
+  @Test(description = "Login page displays when un-logged User clicks on Book ticket tab")
+  public void TC04() {
+    homePage.clickBookTicket();
+    String actualResult = BrowserHelper.getTitle();
```
Comment (thanhlelgg) [inline src/test/java/test_cases/LoginTests.java:59]: don't use page title to verify if a page displayed if there's better option like page header or breadcrumb...
we care more about the UI and less about the browser tab title.

**Review by thanhlelgg — CHANGES_REQUESTED** [review]

File: src/main/java/common/helpers/BrowserHelper.java (hunk 1 above)
Comment (viethuynh140499) [reply src/main/java/common/helpers/BrowserHelper.java]: thanks for letting me know, i removed the handleAlert method and replaced it with the alternate method with two methods: acceptAlert and allowAlert

File: src/test/java/test_cases/LoginTests.java (hunk 2 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/LoginTests.java]: Thanks for letting me know,  I added message for all check point

File: src/test/java/test_cases/LoginTests.java (hunk 3 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/LoginTests.java]: Thanks for letting me know, I replaced `DataHelper.getRandomInvalidPassword ()` with `Constan.INVALID_PASSWORD`

**Comment by viethuynh140499** [comment]
reopen

**Pull request closed without merging**